decidim-core/app/controllers/decidim/locales_controller.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+require_dependency "decidim/application_controller"
+
+module Decidim
+  # A controller to allow users switching their locale.
+  class LocalesController < ApplicationController
+    def create
```

Comment:
Not sure I like this approach. This can most probably be implemented as a before_filter in `ApplicationController`, avoiding a redirection.